server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -529,6 +536,9 @@ private RepositoryData safeRepositoryData(long repositoryStateId, Map<String, Bl
             throw new RepositoryException(metadata.name(), "concurrent modification of the index-N file, expected current generation [" +
                 repositoryStateId + "], actual current generation [" + genToLoad + "]");
         }
+        if (cached != null && cached.v1() == genToLoad && cached.v2() != null) {
```

Comment:
Should `cached.v2()` always be non-null if `cached != null`?

Reply:
Jup now it is.